tools/c7n_mailer/c7n_mailer/utils.py

```diff
@@ -33,3 +32,132 @@ def setup_defaults(config):
+        if 'ProhibitedPolicies' in resource:
+            return "%s %s %s %s" % (
+                resource['LoadBalancerName'],
+                "instances: %d" % len(resource['Instances']),
+                "zones: %d" % len(resource['AvailabilityZones']),
+                "prohibited_policies: %s" % ','.join(
+                    resource['ProhibitedPolicies']))
+        return "%s %s %s" % (
+            resource['LoadBalancerName'],
+            "instances: %d" % len(resource['Instances']),
+            "zones: %d" % len(resource['AvailabilityZones']))
+    elif resource_type == 'redshift':
+        return "%s %s %s" % (
+            resource['ClusterIdentifier'],
+            'nodes:%d' % len(resource['ClusterNodes']),
+            'encrypted:%s' % resource['Encrypted'])
+    elif resource_type == 'emr':
+        return "%s status:%s" % (
+            resource['Id'],
+            resource['Status']['State'])
+    elif resource_type == 'cfn':
+        return "%s" % (
+            resource['StackName'])
+    elif resource_type == 'launch-config':
+        return "%s" % (
+            resource['LaunchConfigurationName'])
+    elif resource_type == 'security-group':
+        name = resource.get('GroupName', '')
+        for t in resource.get('Tags', ()):
+            if t['Key'] == 'Name':
+                name = t['Value']
+        return "%s %s %s inrules: %d outrules: %d" % (
+            name,
+            resource['GroupId'],
+            resource.get('VpcId', 'na'),
+            len(resource.get('IpPermissions', ())),
+            len(resource.get('IpPermissionsEgress', ())))
+    elif resource_type == 'log-group':
+        return "name: %s last_write: %s" % (
+            resource['logGroupName'],
+            resource['lastWrite'])
+    elif resource_type == 'cache-cluster':
+        return "name: %s created: %s status: %s" % (
+            resource['CacheClusterId'],
+            resource['CacheClusterCreateTime'],
+            resource['CacheClusterStatus'])
+    elif resource_type == 'cache-snapshot':
+        return "name: %s cluster: %s source: %s" % (
+            resource['SnapshotName'],
+            resource['CacheClusterId'],
+            resource['SnapshotSource'])
+    elif resource_type == 'redshift-snapshot':
+        return "name: %s db: %s" % (
+            resource['SnapshotIdentifier'],
+            resource['DBName'])
+    elif resource_type == 'ebs-snapshot':
+        return "name: %s date: %s" % (
+            resource['SnapshotId'],
+            resource['StartTime'])
+    elif resource_type == 'subnet':
+        return "%s %s %s %s %s %s" % (
+            resource['SubnetId'],
+            resource['VpcId'],
+            resource['AvailabilityZone'],
+            resource['State'],
+            resource['CidrBlock'],
+            resource['AvailableIpAddressCount'])
+    else:
+        print "Unknown resource type", resource_type
+        return "%s" % format_struct(resource)
```

Comment:
@JohnTheodore FYI - sending users an email to rotate keys will throw Unknown resource type iam-user